src/rosdiscover/cli.py

```diff
@@ -48,33 +46,34 @@ def launch(args) -> None:
     else:
         print(yaml.dump(output, default_flow_style=False))
 
+def nodes_from_yaml(args):
+    with open(args.arch.name, 'r') as f: 
+            data = yaml.safe_load(f) 
+    nodes = []  
+    for d in data: #create the NodeSummary object from a dictionary, append to nodes list
+        read = []
+        write = []
+        same_fields = {'pubs' : [], 'subs' : [], 'uses' : [], 'provides' : [], 
+                        'action-servers' : [], 'action-clients' : []}
+        for i in same_fields:
+            for j in d[i]:
+                same_fields[i].append((j['name'], j['format']))
+        for i in d['reads']:
+            read.append((i['name'], i['dynamic']))
+        for i in d['writes']:
+            write.append(i)
+        obj = NodeSummary(d['name'], d['fullname'], d['namespace'], 
```

Comment:
To avoid unexpected mishandling of arguments, use keyword arguments to build the node summary. (This is also resilient to any possible API changes.)
```
NodeSummary(name=d['name'], ...)
```